### PR TITLE
Added video conversion support

### PR DESCRIPTION
**Pull request by 05Alston**
To try the conversion,
```
from ImageGoNord import NordPaletteFile, GoNord

go_nord = GoNord()

go_nord.convert_video(input_file, palette_name, number_of_frames_in_batch)
```

A higher number of frames per batch means more memory consumption but faster execution as there'll be less parts to merge.

On the first execution, a `nord.npz` color map file will be generated for faster execution.

### CODE REVIEW

File: ImageGoNord/GoNord.py [hunk 1]
```diff
@@ -469,3 +473,232 @@ def save_image_to_file(self, image, path):
...
+        frame_number = 0
+        timestamp = 0
+        batch_dur = frames_per_batch / framerate
+        batch_dur = batch_dur if duration > batch_dur else duration
+        print('####VIDEO INFORMATION#####')
+        print(f'Width: {width}')
+        print(f'Height: {height}')
+        print(f'FPS: {framerate}')
+        print(f'Duration: {duration} s\n')
+        print(f'Processed: {frame_number} / {total_frames} frames')
```
Comment (TheJoin95) [inline ImageGoNord/GoNord.py:689]: Could you remove these printf?

File: ImageGoNord/GoNord.py [hunk 2]
```diff
@@ -469,3 +473,232 @@ def save_image_to_file(self, image, path):
...
+
+        Returns
+        -------
+        None
+            Convert input video and save to disk
+        """
+        # Generate some random unique identifier that is generated for each session for the temporary files.
+        uid = uuid.uuid4()
+        palette = list(self.PALETTE_DATA.values())
+        #TODO: Find a fix for palette_name
```
Comment (TheJoin95) [inline ImageGoNord/GoNord.py:665]: Is this still in todo?

**Comment by TheJoin95** [comment]
Hi, thanks for adding the video support, overall it's looking super!
I added couple of comments on the debug messages.

I see there are some part hardcoded with Nord palette: we would like to extend this also to other palettes, do you think it would be possible?

Could I ask you to add a demo video in your PR description, and also update the readme by specifying this usage (and add the demo video). Thanks!

File: ImageGoNord/GoNord.py (hunk 2 above)
Comment (05Alston) [reply ImageGoNord/GoNord.py]: I'm thinking of adding the palette name as a parameter to the convert_video function itself. Will that be fine?  
The palette_name variable is actually required for name of the `.npz` file mentioned above.

Judging by the interaction with IGN website, it wouldn't be an issue for other palettes as with them it's not possible to deselect individual colors so we could add the palette folder name for them.

**Comment by 05Alston** [comment]
> I see there are some part hardcoded with Nord palette: we would like to extend this also to other palettes, do you think it would be possible?

I've replied to this in the above comment .

> update the readme by specifying this usage (and add the demo video).

Updated the README file

## PS:
For some reason ffmpeg is refusing to convert to gif files directly. I'm looking to fix this issue. Also, for now I've coded the code to only give mp4 as output. I'm working on that is well.

## Demo:
**Original**
luffy

**Converted**

https://github.com/Schrodinger-Hat/ImageGoNord-pip/assets/89850018/8abb8ae9-e639-4d63-bada-41fc955fb456

**Original**

https://github.com/Schrodinger-Hat/ImageGoNord-pip/assets/89850018/793e9f1f-c0a6-42fc-afbb-6eeffe77ca3e

**Converted**

https://github.com/Schrodinger-Hat/ImageGoNord-pip/assets/89850018/2953b727-5d16-4978-af47-4a0d986fc197

**Comment by 05Alston** [comment]
> For some reason ffmpeg is refusing to convert to gif files directly. I'm looking to fix this issue

Fixed this issue as well.

The code to run the above program would be

```
from ImageGoNord import NordPaletteFile, GoNord
go_nord = GoNord()
go_nord.convert_video(
    input_file, 
    "nord",  #color palette name
    _frames_per_batch = 200 #optional parameter
)
```

**Review by TheJoin95 — APPROVED** [review]

**Review by Wabri — APPROVED** [review]
Awesome job

**Comment by TheJoin95** [comment]
We should also update the image-go-nord-cli repository with these changes. I think I'm going to take the image-go-nord web repository, @Wabri @05Alston would you like to take the CLI?

**Pull request merged**